app/controllers/hyku/registrations_controller.rb

```diff
@@ -0,0 +1,15 @@
+module Hyku
+  class RegistrationsController < Devise::RegistrationsController
+    def new
+      return super if Settings.devise.account_signup
+      flash[:alert] = t(:'hyku.account_signup')
+      redirect_to root_path
```

Comment:
You can make this a one liner. `redirect_to(root, alert: t(:'hyku.account_signup'))`  Not a big deal though.